tests/DependencyInjection/ConfigurableRectorImportConfigCallsMergeTest.php

```diff
@@ -31,41 +33,95 @@ protected function setUp(): void
      * @dataProvider provideData()
      * @param array<string, string> $expectedConfiguration
      */
-    public function testMainConfigValues(
-        string $config,
-        int $expectedConfigurationCount,
-        array $expectedConfiguration
-    ): void {
-        $this->bootKernelWithConfigs(RectorKernel::class, [$config]);
+    public function testMainConfigValues(string $config, array $expectedConfiguration): void
+    {
+        $rectorConfigsResolver = new RectorConfigsResolver();
+
+        $configFileInfos = $rectorConfigsResolver->resolveFromConfigFileInfo(new SmartFileInfo($config));
+
+        $this->bootKernelWithConfigs(RectorKernel::class, $configFileInfos);
         $this->renameClassRector = $this->getService(RenameClassRector::class);
 
         $oldToNewClasses = $this->privatesAccessor->getPrivateProperty($this->renameClassRector, 'oldToNewClasses');
 
-        $this->assertCount($expectedConfigurationCount, $oldToNewClasses);
         $this->assertSame($expectedConfiguration, $oldToNewClasses);
     }
 
     public function provideData(): Iterator
     {
         yield [
-            __DIR__ . '/config/main_config_with_only_imports.php', 2, [
+            __DIR__ . '/config/main_config_with_only_imports.php', [
                 'old_2' => 'new_2',
                 'old_1' => 'new_1',
             ],
         ];
 
         yield [
-            __DIR__ . '/config/main_config_with_own_value.php', 3, [
+            __DIR__ . '/config/main_config_with_override_value.php', [
                 'old_2' => 'new_2',
                 'old_1' => 'new_1',
-                'old_3' => 'new_3',
+                'old_4' => 'new_4',
             ],
         ];
 
         yield [
-            __DIR__ . '/config/main_config_with_override_value.php', 2, [
+            __DIR__ . '/config/main_config_with_own_value.php', [
                 'old_2' => 'new_2',
                 'old_1' => 'new_1',
+                'old_4' => 'new_4',
+                'old_3' => 'new_3',
```

Comment:
is 4 and 3 order flipped ?

Reply:
See per commit view > https://github.com/rectorphp/rector/pull/5523/commits/00d3d0834feb7d2a1389f6b59167f33ff018535e

I moved the test so follow the ordering of the files on disk.